apps/src/templates/teacherNavigation/UnitCalendar.tsx

```diff
@@ -15,16 +17,69 @@ const WEEKLY_INSTRUCTIONAL_MINUTES_OPTIONS = [
 export const WEEK_WIDTH = 585;
 
 const UnitCalendar: React.FC = () => {
-  const lessons = useSelector(state => getCurrentUnitData(state)?.lessons);
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false); // it is only loading when you do the fetch
+
   const [weeklyInstructionalMinutes, setWeeklyInstructionalMinutes] =
     useState<string>(WEEKLY_INSTRUCTIONAL_MINUTES_OPTIONS[4].toString());
 
+  const unitName = useSelector(
+    (state: {unitSelection: {unitName: string}}) => state.unitSelection.unitName
+  );
+
+  const unitNameFromProgress = useAppSelector(
+    state => state.progress?.scriptName
+  );
+
+  const hasCalendar = useAppSelector(state => state.calendar?.showCalendar);
+
+  const calendarLessons = useAppSelector(
+    state => state.calendar?.calendarLessons
+  );
+
+  const {userId, userType} = useAppSelector(state => ({
+    userId: state.currentUser.userId,
+    userType: state.currentUser.userType,
+  }));
```

Comment:
```suggestion
  const {userId, userType} = useAppSelector(state => state.currentUser);
```
currentUser is already in this object definiton